plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANAStructureAssistant.java

```diff
@@ -65,7 +65,10 @@ protected void findObjectsByMask(JDBCExecutionContext executionContext, JDBCSess
         GenericSchema parentSchema = parentObject instanceof GenericSchema ? (GenericSchema) parentObject : null;
 
         if (objectType == RelationalObjectType.TYPE_TABLE)
-            findTablesByMask(session, parentSchema, objectNameMask, caseSensitive, maxResults, result);
+        	/* 
+        	 * A TYPE_TABLE is requested in any kind of select statement. Hence it should return tables, views and (public) synonyms
+        	 */
+        	findObjectsByMask(session, parentSchema, objectNameMask, caseSensitive, maxResults, result);

```

Comment:
The object types are choosen in the 'DB metadata' search dialog, so here should not find other types.
If searching without schema might be ok to include synonyms with correct target object type in the result.

A completely different solution would be adding 'Synonym' as searchable object type.

Reply:
Done.